### PR TITLE
Feat/admin pages

### PR DESCRIPTION
**Pull request by WillKocur**
## Changes

| Type | Description |
| - | - |
| Feature | Add archived to country model |
| Feature | Add new and archive actions to table model |
| Feature | Add crud pages for new and show |
| Feature | Add edit and archive url paths to data for frontend component |
| TODO | Update readme to mention that 'archived' must be added to model using the engine |

### Open questions!

1. See TODOs in the code (`TODO: `) - can the routes be accessed better. Sometimes they're being formed quite manually and sometimes using route helpers.
2. The frontend expects object keys to be camel case but some aren't (e.g. edit_url). Not clear to me when the mapping takes place. In the frontend? Note that `legend_on` is not camelcase in the frontend, so may be some issues here.
3. I couldn't get v2 of the FilterableTable component to load in this repo. Errors about babel and node modules... 

### Steps

1. Seed countries
4. View country by going to /countries/:id (try id = 1 but if you've reseeded you may need to run Country.first.id in the rails console to find one to use)
5. Edit it by clicking on the edit button
6. Create a new country by going to /countries/new
7. Archive a country by making a put request 
You may need to add `skip_before_action :verify_authenticity_token` in the first line of the TableController to make this easier and then run
```
curl -X POST -F "archived=[INSERT 0 or 1]" http://localhost:3000/countries/:id/archive
```
You can check if it has archived by looking at the attribute in the rails console.

### Expectations

- You should be able to edit existing countries and create new ones
- You should be able to archive and unarchive countries by making a put request to /countries/:id/archive with params { archived: 0 | 1 }
- Pages should be styles as in the designs

## Links

**Codebase:** [39](https://unep-wcmc.codebasehq.com/projects/editable-loadable-filterable/tickets/39) Need to update tickets to reflect work done.

**Designs:** https://www.figma.com/file/FSlRaQoog7lQR7QRuFYCnS/Database-Designs?node-id=236%3A2669

### CODE REVIEW

File: wcmc_components/lib/wcmc_components/filterable.rb [hunk 1]
```diff
@@ -18,8 +18,21 @@ def as_table_row
 
       # table_page_path returns the 'show' path for the resource
       def table_page_path
-        ''
+        base_path
```
Comment (WillKocur) [inline wcmc_components/lib/wcmc_components/filterable.rb:21]: TODO: Check this is actually correct and won't break things...

File: wcmc_components/lib/wcmc_components/filterable/serializer.rb [hunk 2]
```diff
@@ -35,6 +35,8 @@ def serialize_relation_for_table(relation)
       def serialize_item_for_table(item)
         {
           page_url: item.table_page_path,
```
Comment (WillKocur) [inline wcmc_components/lib/wcmc_components/filterable/serializer.rb:37]: In the frontend it needs pageUrl, but here it's page_url. This is why I was slightly confused. Is something else mapping between the two, or does this need updating to match the frontend?

In which case archive_url and edit_url will also both need to be camelCase.

**Comment by Maxycle** [comment]
Whether I do "archived=[INSERT 0]" or "archived=[INSERT 1]" my console shows "archived":false.

**Comment by WillKocur** [comment]
This could be my bad notation in the PR notes. I was running this curl:

```
curl -X POST -F "archived=1" http://localhost:3000/countries/22/archive
```

Is this the request you were making?

**Comment by WillKocur** [comment]
> 
This could be my bad notation in the PR notes. I was running this curl:

curl -X POST -F "archived=1" http://localhost:3000/countries/22/archive

Is this the request you were making?

**Comment by Maxycle** [comment]
Yes. That works nicely now.

File: wcmc_components/lib/wcmc_components/filterable.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (WillKocur) [reply wcmc_components/lib/wcmc_components/filterable.rb]: Ah yes, good spot! Updated.

**Pull request merged**